gunrock/app/bc/bc_app.cu

```diff
@@ -1,425 +1,313 @@
     {
-        gpu_idx[i] = config -> device_list[i];
-    }
+        int src_num = run_num % num_srcs;
+        VertexT src = srcs[src_num];
+        problem.Reset(src, target);
+        enactor.Reset(src, target);
 
-    // Create streams and MordernGPU context for each GPU
-    streams = new cudaStream_t[num_gpus * num_gpus * 2];
-    context = new ContextPtr[num_gpus * num_gpus];
-    if (!config -> quiet) { printf(" using %d GPUs:", num_gpus); }
-    for (int gpu = 0; gpu < num_gpus; ++gpu)
-    {
-        if (!config -> quiet) { printf(" %d ", gpu_idx[gpu]); }
-        util::SetDevice(gpu_idx[gpu]);
-        for (int i = 0; i < num_gpus * 2; ++i)
-        {
-            int _i = gpu * num_gpus * 2 + i;
-            util::GRError(cudaStreamCreate(&streams[_i]),
-                          "cudaStreamCreate fialed.", __FILE__, __LINE__);
-            if (i < num_gpus)
-            {
-                context[gpu * num_gpus + i] =
-                    mgpu::CreateCudaDeviceAttachStream(gpu_idx[gpu],
-                                                       streams[_i]);
-            }
-        }
+        cpu_timer.Start();
+        enactor.Enact(src);
+        cpu_timer.Stop();
+
+        total_time += cpu_timer.ElapsedMillis();
+        problem.Extract(bc_values[src_num], sigmas[src_num], labels[src_num]);
     }
-    if (!config -> quiet) { printf("\n"); }
 
-    dispatchBC(grapho, graphi, config, data_t, context, streams);
+    enactor.Release(target);
+    problem.Release(target);
+    srcs.clear();
+    return total_time;
 }
 
 /*
- * @brief Simple interface take in CSR arrays as input
- *
- * @param[out] bc_scores   Return BC node centrality per nodes
- * @param[in]  num_nodes   Number of nodes of the input graph
- * @param[in]  num_edges   Number of edges of the input graph
+ * @brief Simple interface take in graph as CSR format
+ * @param[in]  num_nodes   Number of veritces in the input graph
+ * @param[in]  num_edges   Number of edges in the input graph
  * @param[in]  row_offsets CSR-formatted graph input row offsets
  * @param[in]  col_indices CSR-formatted graph input column indices
- * @param[in]  source      Source to begin traverse/computation
+ * @param[in]  edge_values CSR-formatted graph input edge weights
+ * @param[in]  num_runs    Number of runs to perform BC
+ * @param[in]  sources     Sources to begin traverse, one for each run
+ * @param[out] bc_values   Return betweenness centrality values per vertex
+ * @param[out] sigmas      Return sigma of each vertex
+ * @param[out] labels      Return label of each vertex
+ * \return     double      Return accumulated elapsed times for all runs
  */
-void bc(
-    float*     bc_scores,
-    const int  num_nodes,
-    const int  num_edges,
-    const int* row_offsets,
-    const int* col_indices,
-    const int  source)
+template <
+    typename VertexT = int,
+    typename SizeT   = int,
+    typename GValueT = float,
+    typename BCValueT = GValueT>
+float bc(
```

Comment:
This function needs a non-templated version for the c interface, please see the `pr_app.cu`.

Reply:
Added